### PR TITLE
Add dstat to caasp workers

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

File: playbooks/openstack-deploy_caasp.yml [hunk 1]
```diff
@@ -43,3 +43,14 @@
...
+- hosts: caasp-workers
+  vars_files:
+    - "{{ playbook_dir }}/../vars/common-vars.yml"
+  tasks:
+    - import_role:
+        name: dstat
+        tasks_from: main
+      when: lookup('env', 'SOCOK8S_DEPLOY_DSTAT') | default(False)
+      tags:
+        - yeah
```
Comment (jsuchome) [inline playbooks/openstack-deploy_caasp.yml]: Yeah, why not!

Comment (Itxaka) [reply playbooks/openstack-deploy_caasp.yml]: lmao, leftover of local testing, will remove!

**Comment by Itxaka** [comment]
now, we could also generate graphs based on the output for easy consumption, no idea how, probably with gnuplot but Im a total ignorant of that

File: playbooks/generic-gather_dstat_output.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+- hosts: caasp-workers
+  gather_facts: yes
+  vars_files:
+    - "{{ playbook_dir }}/../vars/common-vars.yml"
+  tasks:
+    - import_role:
+        name: dstat
+        tasks_from: fetch-data
+      when: lookup('env', 'SOCOK8S_DEPLOY_DSTAT') | default(False)
```
Comment (evrardjp) [inline playbooks/generic-gather_dstat_output.yml]: ```default(False, True)``` when empty, you should make that false.

File: playbooks/kvm-deploy_caasp.yml [hunk 3]
```diff
@@ -43,3 +43,12 @@
...
         image_username: "{{ skuba_ci_terraform_libvirt_image_username }}"
+
+- hosts: caasp-workers
+  vars_files:
+    - "{{ playbook_dir }}/../vars/common-vars.yml"
+  tasks:
+    - import_role:
+        name: dstat
+        tasks_from: main
+      when: lookup('env', 'SOCOK8S_DEPLOY_DSTAT') | default(False)
```
Comment (evrardjp) [inline playbooks/kvm-deploy_caasp.yml]: ditto.

File: playbooks/kvm-deploy_caasp.yml [hunk 4]
```diff
@@ -43,3 +43,12 @@
       vars:
         skuba_cluster_basedir: "{{ socok8s_workspace }}"
         image_username: "{{ skuba_ci_terraform_libvirt_image_username }}"
+
+- hosts: caasp-workers
+  vars_files:
+    - "{{ playbook_dir }}/../vars/common-vars.yml"
+  tasks:
+    - import_role:
```
Comment (evrardjp) [inline playbooks/kvm-deploy_caasp.yml]: just make it a role, instead of tasks -import_role ?

File: playbooks/openstack-deploy_caasp.yml [hunk 5]
```diff
@@ -43,3 +43,12 @@
...
         image_username: "{{ skuba_ci_terraform_openstack_image_username }}"
+
+- hosts: caasp-workers
+  vars_files:
+    - "{{ playbook_dir }}/../vars/common-vars.yml"
+  tasks:
+    - import_role:
+        name: dstat
+        tasks_from: main
+      when: lookup('env', 'SOCOK8S_DEPLOY_DSTAT') | default(False)
```
Comment (evrardjp) [inline playbooks/openstack-deploy_caasp.yml]: ditto

File: playbooks/generic-gather_dstat_output.yml (hunk 2 above)
Comment (Itxaka) [reply playbooks/generic-gather_dstat_output.yml]: done!

File: playbooks/kvm-deploy_caasp.yml (hunk 4 above)
Comment (Itxaka) [reply playbooks/kvm-deploy_caasp.yml]: done!

File: playbooks/kvm-deploy_caasp.yml (hunk 3 above)
Comment (Itxaka) [reply playbooks/kvm-deploy_caasp.yml]: done!

File: playbooks/openstack-deploy_caasp.yml (hunk 5 above)
Comment (Itxaka) [reply playbooks/openstack-deploy_caasp.yml]: done!

**Comment by Itxaka** [comment]
For some reason it wont work properly, I dont know why because on my manual env deployed it does...

File: playbooks/roles/dstat/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+- name: Check if dstat is already running
+  command: pgrep dstat
+  register: _dstat_running
+  failed_when: false
+  changed_when: false
+
+
+- name: Run dstat in the background
+  shell: nohup dstat {{ dstat_options }} --nocolor --output {{ dstat_output_csv }} {{ dstat_delay }} </dev/null > {{ dstat_output_log }} 2>&1 &
+  poll: 0
```
Comment (evrardjp) [inline playbooks/roles/dstat/tasks/main.yml]: poll is not necessary.

File: playbooks/roles/dstat/defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+# -n: network stats
+# -p: proc stats
+# -r: io stats
+# -g: page stats
+# -i: interrupt stats
+# -l: load stats
+dstat_options: "-tcmdnl"
+# delay is the delay in seconds between each update
+dstat_delay: "5"
+dstat_output_csv: "{{ ansible_env.HOME }}/dstat.csv"
```
Comment (evrardjp) [inline playbooks/roles/dstat/defaults/main.yml]: ansible_facts['user_dir'] is probably better.

**Comment by Itxaka** [comment]
Green and artifacts archived, nice! https://ci.suse.de/blue/organizations/jenkins/cloud-socok8s-jobs%2Fsocok8s-integration/detail/PR-726/24/artifacts

File: playbooks/roles/dstat/defaults/main.yml (hunk 7 above)
Comment (Itxaka) [reply playbooks/roles/dstat/defaults/main.yml]: are you ok with keeping it this way for now? 

BTW I used that so its in line with other places where we use the HOME like this

Comment (evrardjp) [reply playbooks/roles/dstat/defaults/main.yml]: depends if it's the remote user or the local user :)

**Pull request merged**